File: src/main/java/toker/panel/entity/PanelUserAuthorityAssignment.java
package toker.panel.entity;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "panel_user_authority_assignment")
public class PanelUserAuthorityAssignment {
    private Integer id;
    private PanelUser panelUser;
    private Server server;
    private PanelUserAuthority authority;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @ManyToOne
    @JoinColumn(name = "panel_user_id", referencedColumnName = "id", nullable = false)
    public PanelUser getPanelUser() {
        return panelUser;
    }

    public void setPanelUser(PanelUser panelUser) {
        this.panelUser = panelUser;
    }

    @ManyToOne
    @JoinColumn(name = "server_id", referencedColumnName = "id", nullable = false)
    public Server getServer() {
        return server;
    }

    public void setServer(Server server) {
        this.server = server;
    }

    @ManyToOne
    @JoinColumn(name = "authority_id", referencedColumnName = "id", nullable = false)
    public PanelUserAuthority getAuthority() {
        return authority;
    }

    public void setAuthority(PanelUserAuthority authority) {
        this.authority = authority;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        PanelUserAuthorityAssignment that = (PanelUserAuthorityAssignment) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/main/java/toker/panel/entity/PanelUserAuthorityAssignment.kt
package toker.panel.entity

import com.fasterxml.jackson.annotation.JsonView
import javax.persistence.*

@Entity
@Table(name = "panel_user_authority_assignment")
class PanelUserAuthorityAssignment(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", nullable = false)
        var id: Int? = null,

        @ManyToOne
        @JoinColumn(name = "panel_user_id", referencedColumnName = "id", nullable = false)
        @JsonView(View.PanelUser::class)
        var panelUser: PanelUser? = null,

        @ManyToOne
        @JoinColumn(name = "server_id", referencedColumnName = "id", nullable = false)
        @JsonView(View.Server::class)
        var server: Server? = null,

        @ManyToOne
        @JoinColumn(name = "authority_id", referencedColumnName = "id", nullable = false)
        @JsonView(View.Authority::class)
        var authority: PanelUserAuthority? = null
) {
    interface View {
            interface PanelUser
            interface Server
            interface Authority
    }
}